packages/cli/src/lib/utils/iterator.spec.ts

```diff
@@ -0,0 +1,81 @@
+import {inspect} from 'util';
+import {consumeIterator} from './iterator';
+
+class Deferred {
+  public promise: Promise<void>;
+  private _resolve!: Function;
+  public constructor() {
+    this.promise = new Promise((resolve) => {
+      this._resolve = resolve;
+    });
+  }
+
+  public async resolve() {
+    this._resolve();
+    return this.promise;
+  }
+}
+
+describe('iterator', () => {
```

Comment:
That's the best/easiest way to test the promise concurrency...
I am not able to test the concurrency via the `source:push:add` command since I did not find any way to run any `expect` while there are still pending promises going on.

Reply:
Nice one! You isolated the behavior to test, this should offer 'good enough' coverage ;)